diff --git a/luci-app-ssr-plus/Makefile b/luci-app-ssr-plus/Makefile
index 43ae97b..890888c 100644
--- a/luci-app-ssr-plus/Makefile
+++ b/luci-app-ssr-plus/Makefile
@@ -48,6 +48,7 @@ LUCI_DEPENDS:= \
 
 
 define Package/$(PKG_NAME)/config
+if PACKAGE_$(PKG_NAME)
 
 choice
 	prompt "Transparent Proxy Backend"
@@ -171,6 +172,7 @@ config PACKAGE_$(PKG_NAME)_INCLUDE_ShadowsocksR_Libev_Server
 	bool "Include ShadowsocksR Libev Server"
 	default n
 
+endif
 endef
 
 
